### ISSUE TITLE
SlevomatCodingStandard.PHP.UselessParentheses false-positive with closures

### ISSUE THREAD

**Issue by EvgenyOrekhov** [slevomat/coding-standard#443]
```php
class C
{
    public $closure;

    public function __construct()
    {
        $this->closure = function () {
            echo 123;
        };

        ($this->closure)();
    }
}
```

Expected result: no warnings

Actual result:

> Useless parentheses.

You can't omit the parens in `($this->closure)()`, since it would become a method call `$this->closure()`.

Slevomat Coding Standard version: 4.7.1

**Comment by kukulich** [comment]
It is unfortunatelly impossible to detect.

**Comment by kukulich** [comment]
Or maybe...

**Comment by EvgenyOrekhov** [comment]
@kukulich `$a->b()` and `($a->b)()` are totally different ways to call different things, they are not interchangeable. If you have a closure, you can call it only like this: `($a->b)()`, and if you have a method, you can call it only like this: `$a->b()`. You can't call a method by adding an extra parens around it, it will be a fatal error `Function name must be a string`.

**Comment by EvgenyOrekhov** [comment]
@kukulich In other words, if you have a method, then the parens in `($a->b)()` are not "extra", they are "fatal error", they are totally wrong. So I'd say there is no need to catch this case by the sniffer.

**Comment by kukulich** [comment]
Thank your for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/37cd2acb69374f875c618c9cce68af1737618df3
